testing/trino-testing/src/main/java/io/trino/testing/AbstractTestEngineOnlyQueries.java

```diff
@@ -6694,6 +6694,47 @@ SELECT a(10)
                 .hasMessage("line 3:8: Recursive language functions are not supported: a(integer):integer");
     }
 
+    // regression test for https://github.com/trinodb/trino/issues/19997
+    @Test
+    public void testIssue19997()
+    {
+        assertThat(query("""
```

Comment:
There aren't any more generic tests for dereferences that could be updated?

Reply:
i didn't find function test with row dereferences. see https://github.com/trinodb/trino/pull/20181#discussion_r1433186980